bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeDiscoveryService.java

```diff
@@ -0,0 +1,185 @@
+ *
+ * @author Miguel Alvarez - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class, configurationPid = "discovery.androiddebugbridge")
+public class AndroidDebugBridgeDiscoveryService extends AbstractDiscoveryService {
+    static final int TIMEOUT_MS = 60000;
+    private static final long DISCOVERY_RESULT_TTL_SEC = 300;
+    public static final String LOCAL_INTERFACE_IP = "127.0.0.1";
+    public static final int MAX_RETRIES = 2;
+    private final Logger logger = LoggerFactory.getLogger(AndroidDebugBridgeDiscoveryService.class);
+    private boolean discoveryRunning = false;
+    private ConfigurationAdmin admin;
+
+    @Activate
+    public AndroidDebugBridgeDiscoveryService(@Reference ConfigurationAdmin admin) {
+        super(SUPPORTED_THING_TYPES, TIMEOUT_MS, false);
+        this.admin = admin;
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("scan started: searching android devices");
+        discoveryRunning = true;
+        Enumeration<NetworkInterface> nets;
+        Dictionary<String, Object> props = getConfig();
+        if (props == null) {
+            return;
+        }
+        var discoveryPort = Integer.parseInt((String) props.get("discoveryPort"));
+        var discoveryReachableMs = Integer.parseInt((String) props.get("discoveryReachableMs"));
+        var discoveryIpRangeMin = Integer.parseInt((String) props.get("discoveryIpRangeMin"));
+        var discoveryIpRangeMax = Integer.parseInt((String) props.get("discoveryIpRangeMax"));
+        try {
+            nets = NetworkInterface.getNetworkInterfaces();
+            for (NetworkInterface netint : Collections.list(nets)) {
+                Enumeration<InetAddress> inetAddresses = netint.getInetAddresses();
+                for (InetAddress inetAddress : Collections.list(inetAddresses)) {
+                    if (!discoveryRunning)
+                        break;
+                    if (!(inetAddress instanceof Inet4Address))
+                        continue;
+                    if (inetAddress.getHostAddress().equals(LOCAL_INTERFACE_IP))
+                        continue;
+                    String[] ipParts = inetAddress.getHostAddress().split("\\.");
+                    for (var i = discoveryIpRangeMin; i <= discoveryIpRangeMax; i++) {
+                        if (!discoveryRunning)
+                            break;
+                        ipParts[3] = Integer.toString(i);
+                        var currentIp = String.join(".", ipParts);
+                        try {
+                            var currentAddress = InetAddress.getByName(currentIp);
+                            logger.debug("address: {}", currentIp);
+                            if (currentAddress.isReachable(discoveryReachableMs)) {
+                                logger.debug("Reachable ip: {}", currentIp);
+                                var retries = 0;
```

Comment:
Please don't use `var` for primitives, it just leads to possible confusion.
```suggestion
                                int retries = 0;
```